sdk/core/azure-core-amqp/src/amqp/management.cpp

```diff
@@ -78,41 +60,60 @@ namespace Azure { namespace Core { namespace Amqp { namespace _internal {
 }}}} // namespace Azure::Core::Amqp::_internal
 
 namespace Azure { namespace Core { namespace Amqp { namespace _detail {
-  ManagementImpl::ManagementImpl(
+  ManagementClientImpl::ManagementClientImpl(
       std::shared_ptr<SessionImpl> session,
       std::string const& managementEntityPath,
-      Azure::Core::Amqp::_internal::ManagementOptions const& options,
-      Azure::Core::Amqp::_internal::ManagementEvents* managementEvents)
-      : m_options{options}, m_session{session}, m_eventHandler{managementEvents}
+      Azure::Core::Amqp::_internal::ManagementClientOptions const& options,
+      Azure::Core::Amqp::_internal::ManagementClientEvents* managementEvents)
+      : m_options{options}, m_session{session}, m_eventHandler{managementEvents},
+        m_managementEntityPath{managementEntityPath}
   {
-    /** Authentication needs to happen *before* the management object is created. */
-    m_session->AuthenticateIfNeeded(managementEntityPath + "/" + m_options.ManagementNodeName, {});
+  }
+  ManagementClientImpl::~ManagementClientImpl() noexcept { m_eventHandler = nullptr; }
 
-    m_management.reset(amqp_management_create(*session, m_options.ManagementNodeName.c_str()));
-    if (options.EnableTrace)
+  void ManagementClientImpl::CreateManagementClient()
+  {
+
+    m_management.reset(amqp_management_create(*m_session, m_options.ManagementNodeName.c_str()));
+    if (!m_management)
     {
-      amqp_management_set_trace(m_management.get(), options.EnableTrace);
+      throw std::runtime_error("Could not create management object.");
```

Comment:
Is this enough details for debugging the failure?

Reply:
No, but the only additional information I would have is the result of the call to amqp_management_create, which is the line number of the failure in the amqp_management.c source file. That isn't very helpful either.

Reply:
Theoretically, the uAMQP code logs error information in the trace before returning an error, so that information is there if you enable tracing. But it isn't consistent :(.